src/main/java/net/sigmarik/abilitymod/mixin/PlayerEntityMixin.java

```diff
@@ -79,11 +85,48 @@ private void tickBoatMagnet() {
         }
     }
 
+    private void tickLightBurn() {
+        if (ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_DAMAGED_BY_LIGHT)) {
+            World world = this.getWorld();
+            ItemStack helmet = this.getInventory().getArmorStack(3);
+            long time = world.getTimeOfDay() % 24000;
+            int lightLevel = world.getLightLevel(LightType.SKY, this.getBlockPos());
+            if (!(lightLevel <= 11 ||
+                    !world.isSkyVisible(this.getBlockPos()) || this.isTouchingWaterOrRain() || this.isInLava() ||
+                    this.hasStatusEffect(StatusEffect.byRawId(AbilityMod.FIRE_RESISTANCE_STATUS_EFFECT)) ||
```

Comment:
`hasStatusEffect(StatusEffects.FIRE_RESISTANCE);`
There is no need to add magical numbers that can change in the next update of the game. You can just use existing values.